how-to/workspace-platform-starter/client/src/framework/workspace/notification-client.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ *
+ */
+export class NotificationClient implements NotificationClientInterface {
+	private readonly _options: NotificationClientOptions;
+
+	private readonly _idPrefix: string;
+
+	private readonly _platformId: string;
+
+	private _listenerRegister: { id: string; listener: never; wrappedListener: never }[] = [];
+
+	/**
+	 * Notification options.
+	 * @param options Notification options.
+	 * @param platformId The id of the current platform this client will belong to.
+	 */
+	constructor(options: NotificationClientOptions, platformId: string) {
+		this._options = options;
+		this._idPrefix = options.idPrefix ?? `${options.id}-`;
+		this._platformId = platformId;
+	}
+
+	/**
+	 * Create a notification.
+	 * @param options the notification to create.
+	 * @param creationOptions creationOptions for the notification.
+	 * @returns created notification.
+	 */
+	public async create<T extends NotificationOptions>(
+		options: T,
+		creationOptions?: NotificationCreationOptions
+	): Promise<Notification<T>> {
+		if (!this.hasId(options.id)) {
+			options.id = this.createId(options.id);
+		}
+		if (this._options.enforceIcon && !isEmpty(this._options.icon)) {
+			options.icon = this._options.icon;
+		}
+
+		if (this._options.includeInPlatform) {
+			options.platform = this._platformId;
+		} else if (options.platform === this._platformId) {
+			options.platform = undefined;
+		}
+
+		return notificationCreate(options, creationOptions);
+	}
+
+	/**
+	 * Allows the updating of an existing notification.
+	 * @param options the notification to update.
+	 * @returns the updated notification.
+	 */
+	public async update<T extends UpdatableNotificationOptions>(options: T): Promise<Notification> {
+		if (!this.hasId(options.id)) {
+			options.id = this.createId(options.id);
+		}
+
+		return notificationUpdate(options);
+	}
+
+	/**
+	 * Get all notifications.
+	 * @returns all the notifications.
+	 */
+	public async getAll(): Promise<Notification[]> {
+		const notifications = await notificationGetAll();
+		return notifications.filter((entry) => entry.id.startsWith(this._idPrefix));
+	}
+
+	/**
+	 * Clears a specific notification.
+	 * @param id the id of the notification to clear.
+	 * @returns whether the notification was cleared.
+	 */
+	public async clear(id: string): Promise<boolean> {
+		const notificationToClear = this.createId(id);
+		return notificationClear(notificationToClear);
+	}
+
+	/**
+	 * Clear all notifications.
+	 * @returns the number of notifications that was cleared.
+	 */
+	public async clearAll(): Promise<number> {
+		const notifications = await this.getAll();
+		let count = 0;
+		for (const notification of notifications) {
+			const removed = await this.clear(notification.id);
+			if (removed) {
+				count++;
+			}
+		}
+		return count;
+	}
+
+	/**
+	 * Shows the notification center.
+	 * @param options the options to show.
+	 */
+	public async show(options?: ShowOptions | undefined): Promise<void> {
```

Comment:
Why is this optional and have undefined as a type

Reply:
Updated.